packages/audiofileplayer/example/lib/main.dart

```diff
@@ -79,12 +110,52 @@ class _MyAppState extends State<MyApp> {
             ],
           ));
 
-  // convert double seconds to minutes:seconds
+  /// Convert double seconds to String minutes:seconds.
   static String _stringForSeconds(double seconds) {
     if (seconds == null) return null;
     return '${(seconds ~/ 60)}:${(seconds.truncate() % 60).toString().padLeft(2, '0')}';
   }
 
+  /// Listener for transport control events from the OS.
```

Comment:
Ah.. I did not realize that the background operations are being controlled from Dart too. In that case, it makes sense that you give full control of icons to Dart as well.

Are you assuming that the app is running from a foreground service or from the application? I would think Android could kill all activities and leave only the service running in which case you will not get these callbacks.

Reply:
That (Activity and Dart VM will be killed while backgrounded) was my initial assumption as well. However, it appears that an Activity that is connected to a MediaBrowserService will not be killed in the background. (Even weirder, this remains the case even if the Foreground Service is stopped). 

That means the MediaPlayers can stay in the plugin code, not the Service code, and the Dart code can contain all the business logic.

Reply:
Yikes. That makes me nervous given Android's propensity to change behavior from version to version (and the fact that they have been tightening on battery consuming operations in the past 2 releases).

However, it is indeed much nicer to have this logic in Dart as opposed to native so LGTM.

Reply:
Agreed on the Yikes; I'll keep an eye out for this breaking...I've seen other projects based on this assumption, so I'm sure people will notice.